resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/SparkPodInitContainerBootstrapSuite.scala

```diff
@@ -16,151 +16,73 @@
 
+import scala.collection.JavaConverters._
 import org.apache.spark.SparkFunSuite
-import org.apache.spark.deploy.kubernetes.constants._
 
 class SparkPodInitContainerBootstrapSuite extends SparkFunSuite with BeforeAndAfter {
-  private val OBJECT_MAPPER = new ObjectMapper()
   private val INIT_CONTAINER_IMAGE = "spark-init:latest"
   private val DOCKER_IMAGE_PULL_POLICY = "IfNotPresent"
   private val JARS_DOWNLOAD_PATH = "/var/data/spark-jars"
   private val FILES_DOWNLOAD_PATH = "/var/data/spark-files"
   private val DOWNLOAD_TIMEOUT_MINUTES = 5
   private val INIT_CONTAINER_CONFIG_MAP_NAME = "spark-init-config-map"
   private val INIT_CONTAINER_CONFIG_MAP_KEY = "spark-init-config-map-key"
-  private val ADDED_SUBMITTED_DEPENDENCY_ENV = "ADDED_SUBMITTED_DEPENDENCY"
-  private val ADDED_SUBMITTED_DEPENDENCY_ANNOTATION = "added-submitted-dependencies"
   private val MAIN_CONTAINER_NAME = "spark-main"
-  private val TRUE = "true"
 
-  private val submittedDependencyPlugin = new InitContainerResourceStagingServerSecretPlugin {
-    override def addResourceStagingServerSecretVolumeToPod(basePod: PodBuilder)
-        : PodBuilder = {
-      basePod.editMetadata()
-        .addToAnnotations(ADDED_SUBMITTED_DEPENDENCY_ANNOTATION, TRUE)
-        .endMetadata()
-    }
-
-    override def mountResourceStagingServerSecretIntoInitContainer(container: ContainerBuilder)
-        : ContainerBuilder = {
-      container
-        .addNewEnv()
-          .withName(ADDED_SUBMITTED_DEPENDENCY_ENV)
-          .withValue(TRUE)
-          .endEnv()
-    }
-  }
-
-  test("Running without submitted dependencies adds init-container with volume mounts.") {
-    val bootstrappedPod = bootstrapPodWithoutSubmittedDependencies()
-    val podAnnotations = bootstrappedPod.getMetadata.getAnnotations.asScala
-    assert(podAnnotations.contains(INIT_CONTAINER_ANNOTATION))
-    val initContainers = OBJECT_MAPPER.readValue(
-        podAnnotations(INIT_CONTAINER_ANNOTATION), classOf[Array[Container]])
-    assert(initContainers.length === 1)
-    val initContainer = initContainers.head
-    val initContainerVolumeMounts = initContainer.getVolumeMounts.asScala.map {
-      mount => (mount.getName, mount.getMountPath)
-    }.toMap
-    val expectedInitContainerVolumeMounts = Map(
-      INIT_CONTAINER_PROPERTIES_FILE_VOLUME -> INIT_CONTAINER_PROPERTIES_FILE_DIR,
-      INIT_CONTAINER_DOWNLOAD_JARS_VOLUME_NAME -> JARS_DOWNLOAD_PATH,
-      INIT_CONTAINER_DOWNLOAD_FILES_VOLUME_NAME -> FILES_DOWNLOAD_PATH)
-    assert(initContainerVolumeMounts === expectedInitContainerVolumeMounts)
+  private val sparkPodInit = new SparkPodInitContainerBootstrapImpl(
+    INIT_CONTAINER_IMAGE,
+    DOCKER_IMAGE_PULL_POLICY,
+    JARS_DOWNLOAD_PATH,
+    FILES_DOWNLOAD_PATH,
+    DOWNLOAD_TIMEOUT_MINUTES,
+    INIT_CONTAINER_CONFIG_MAP_NAME,
+    INIT_CONTAINER_CONFIG_MAP_KEY
+  )
+  private val returnedPodWithCont = sparkPodInit.bootstrapInitContainerAndVolumes(
```

Comment:
Sharing fields is nice for the object under test, but for the functionality we specifically are testing, it's more idiomatic to put the method calls into the test.